packages/amplify-category-function/src/provider-utils/awscloudformation/utils/addLayerToFunctionUtils.ts

```diff
@@ -63,23 +64,38 @@ export const askLayerSelection = async (
   layerSelections = layerSelections.filter(selection => selection !== provideExistingARNsPrompt);
 
   for await (let layerName of layerSelections) {
-    const currentSelectionDefaults = filterProjectLayers(previousSelections).find(sel => sel.resourceName === layerName);
-    const currentVersion = currentSelectionDefaults ? currentSelectionDefaults.version.toString() : undefined;
     const layerVersions = await loadLayerDataFromCloud(context, layerName);
-    const layerVersionPrompt: ListQuestion = {
-      type: 'list',
-      name: 'versionSelection',
-      message: versionSelectionPrompt(layerName),
-      choices: layerVersions.map(layerVersion => layerVersion.Version.toString()),
-      default: currentVersion,
-    };
+    const layerVersionArrPrompt = layerVersions.map(layerVersion => layerVersion.Version.toString());
+    // skip asking version for a new layer
+    if (Array.isArray(layerVersionArrPrompt) && layerVersionArrPrompt.length) {
+      layerVersionArrPrompt.unshift(defaultLayerVersionPrompt);
+      const layerVersionPrompt: ListQuestion = {
+        type: 'list',
+        name: 'versionSelection',
+        message: versionSelectionPrompt(layerName),
+        choices: layerVersionArrPrompt,
+        default: defaultLayerVersionPrompt,
+      };
+      const versionSelection = (await inquirer.prompt(layerVersionPrompt)).versionSelection;
+      const isLatestVersionSelected = versionSelection.toString() === defaultLayerVersionPrompt ? true : false;
+      const selectedVersion = versionSelection.toString() === defaultLayerVersionPrompt ? layerVersionArrPrompt[1] : versionSelection;
```

Comment:
From what I can tell, this just pins the latest known version. We need to depend on the latest output of the layer CFN using `Fn::GetAtt`.

Reply:
Yeah , for latest layer , we are using Ref and the parameters is populated using dependsOn section during amplify push.
The version used here is to save ```lambdaLayers``` params in ```function-parameters.json```

Reply:
Thanks for the clarification 👍. Can we store the `defaultLayerVersionPrompt` instead of the version number or will it break things?

Reply:
done!

Reply:
done!